### ISSUE TITLE
expo-permission@1.1.0 on android 6.0.1 causes crash on Permissions.askAsync

### ISSUE THREAD

**Issue by jasekiw** [expo/expo#2196]
### Environment

```
Environment:
  OS: Windows 10
  Node: 8.11.1
  Yarn: 0.21.3
  npm: 5.8.0
  Watchman: Not Found
  Xcode: N/A
  Android Studio: Version  3.0.0.0 AI-171.4408382

Packages: (wanted => installed)
  expo: ^29.0.0 => 29.0.0
  react: 16.3.1 => 16.3.1
  react-native: https://github.com/expo/react-native/archive/sdk-29.0.0.tar.gz => 0.55.4

Diagnostics report:
  https://exp-xde-diagnostics.s3.amazonaws.com/jasekiw-e8fb1bed-a1bc-4dd1-9741-c061b24feacf.tar.gz
```

Target: `Android 6.0.1`

Package: `expo-permissions@1.1.0`

### Steps to Reproduce

Follow these steps or clone the reproduceable demo.

1. Setup the project
```
exp init test-project
cd test-project
npm install # this is required for the dependencies of expo to get updated to the latest version.
```
2. Open the app and you will get another issue that I will post. add global.self = gloabl at the top to get rid of it.
3. add `import {Permissions} from 'expo';`
4. Ask for a permission using ` const {status} = await Permissions.askAsync(Permissions.CAMERA);`

### Expected Behavior

The App will ask for the permission

### Actual Behavior
An error displays in the warning window.

`Unhandled promise rejection: Error: Argument of an incompatible class: class java.util.ArrayList cannot be passed as an argument to parameter expecting class java.lang.String.`

### Reproducible Demo

https://github.com/jasekiw/expo-permission-issue

Follow the instructions in the readme.

**Comment by bbarthec** [comment]
Hello @jasekiw! I've just tested your scenario against sdks 29 and 30 and both worked.
First of all please ensure that each of these are up-to-date:
- `expo-cli`
- Expo client app
- node_modules

The problem you're reporting is only possible when you have mismatch between `expo` version in `package.json` and `sdkVersion` in `app.json`.
I've cloned your repo and tested it against clean environment and everything went smoothly.
The error message that you're showed with comes from following situation:
- you have defined `sdkVersion` = `29.0.0`
- your `node-modules` contains `expo` package with version `30.0.0` or above
- your app is now trying to use `js` part from version `30` along with `native` part from version `29`
- between these version we introduced some api breaking changes that would prevent each other to communicate (for your safety always pay attention to keeping versioning aligned)
- for further explanation see: [v30 docs](https://docs.expo.io/versions/v30.0.0/sdk/permissions#expopermissionsgetasyncpermissiontypes) vs [v29 docs](https://docs.expo.io/versions/v29.0.0/sdk/permissions#expopermissionsgetasynctype)

tl;dr pay attention to versions aligning between what's inside `node_modules` (resulting from `package.json`) and `app.json`
